fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheAnalyzer.java

```diff
@@ -697,15 +700,29 @@ private CacheTable buildCacheTableForOlapScanNode(OlapScanNode node) {
         CatalogIf catalog = database.getCatalog();
         ScanTable scanTable = new ScanTable(
                 new FullTableName(catalog.getName(), database.getFullName(), olapTable.getName()),
-                olapTable.getVisibleVersionTime(), olapTable.getVisibleVersion());
+                olapTable.getVisibleVersion());
         scanTables.add(scanTable);
+        if (Config.isCloudMode()) {
+            // get version of related partitions in batch
```

Comment:
How about extracting this operation into `OlapTable`, just like `OlapTable.selectNonEmptyPartitionIds`?

Reply:
Ops! On second thought, I think can give it a try.